code/game/objects/structures/ladders.dm

```diff
@@ -77,38 +78,73 @@
 		return
 
 	var/ladder_dir_name
-	var/obj/structure/ladder/ladder_dest
-	if(up && down)
-		ladder_dest = lowertext(show_radial_menu(user, src, direction_selection, require_near = TRUE))
-		if(ladder_dest == "up")
-			ladder_dest = up
-			ladder_dir_name = ("up")
-		if(ladder_dest == "down")
-			ladder_dest = down
-			ladder_dir_name = ("down")
+
+	if(!length(direction_selection))
+		direction_selection = get_ladder_images()
+
+	if(length(direction_selection) > 1)
+		var/selected_ladder_dest = lowertext(show_radial_menu(user, src, direction_selection, require_near = TRUE))
+		var/static/regex/ladder_regex = regex(@"(up|down) \(x([0-9])+\)")
+
+		if(!selected_ladder_dest)
+			return
+
+		if(selected_ladder_dest == "up" || selected_ladder_dest == "down")
+			ladder_dir_name = selected_ladder_dest
+
+		if(!ladder_dir_name)
```

Comment:
Selecting to go up once just doesnt work Getting on midship and ladders dont work at all. returns on line 111

Reply:
tfw the return is unintended by one tab